app/src/main/java/de/tum/in/tumcampusapp/component/tumui/calendar/CalendarActivity.java

```diff
@@ -406,23 +406,38 @@ private void deleteCalendarFromGoogle() {
     @Override
     public List<WeekViewDisplayable> onMonthChange(int newYear, int newMonth) {
         // Populate the week view with the events of the month to display
-        List<WeekViewDisplayable> events = new ArrayList<>();
-
         DateTime begin = new DateTime().withDate(newYear, newMonth, 1);
-
         int daysInMonth = begin.dayOfMonth()
                                .getMaximumValue();
-
         DateTime end = new DateTime().withDate(newYear, newMonth, daysInMonth);
+        return filterCalendarItems(calendarController.getFromDbBetweenDates(begin, end));
+    }
 
-        List<CalendarItem> calendarItems = calendarController.getFromDbBetweenDates(begin, end);
+    /**
+     * Handles cancelled events and events with multiple locations.
+     */
+    private List<WeekViewDisplayable> filterCalendarItems(List<CalendarItem> calendarItems) {
```

Comment:
Since we're not only filtering the calendar items, but filtering, merging *and* transforming the Items to `WeekViewDisplayable` the naming of the function isn't appropriate anymore.

To avoid complex functions, I'd propose to split this into three different functions:
`List<WeekViewDisplayable> prepareWeekViewDisplayables(List<CalendarItem>)`
which internally calls:
`List<CalendarItem> filterCanceledItems(List<CalendarItem>)` and
`List<CalendarItem> mergeSimilarItems(List<CalendarItem>)`